src/compiler/transformers/decorators-to-static/convert-decorators.ts

```diff
@@ -143,3 +159,198 @@ const filterDecorators = (
+ * and about the `useDefineForClassFields` TypeScript option which lets you
+ * opt-in to the old output:
+ *
+ * https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#the-usedefineforclassfields-flag-and-the-declare-property-modifier
+ *
+ * For our use-case, however, the ES2022+ behavior doesn't work, since we need
+ * to be able to define getters and setters on these fields. We could require
+ * that the TypeScript configuration used for Stencil have the
+ * `useDefineForClassFields` setting set to `false`, but that would have the
+ * undesirable side-effect that class fields which are _not_
+ * decorated with a Stencil decorator would also be instantiated in the
+ * constructor.
+ *
+ * So instead, we take matters into our own hands. When we encounter a class
+ * field which is decorated with a Stencil decorator we remove it from the
+ * class and add a statement to the constructor to instantiate it with the
+ * correct default value.
+ *
+ * **Note**: this function will modify a constructor if one is already present on
+ * the class or define a new one otherwise.
+ *
+ * @param classNode a TypeScript AST node for a Stencil component class
+ * @param classMembers the class members that we need to update
+ * @returns a list of updated class elements which can be inserted into the class
+ */
+function handleClassFields(classNode: ts.ClassDeclaration, classMembers: ts.ClassElement[]): ts.ClassElement[] {
+  const statements = [];
+  const updatedClassMembers = [];
+
+  for (const member of classMembers) {
+    const currentDecorators = member.decorators;
+    const newDecorators = filterDecorators(member, MEMBER_DECORATORS_TO_REMOVE);
+    if (currentDecorators !== newDecorators && ts.isPropertyDeclaration(member)) {
+      // this is a class field that we'll need to handle, so lets push a statement for
+      // initializing the value onto our statements list
+      statements.push(
+        ts.factory.createExpressionStatement(
+          ts.factory.createBinaryExpression(
+            ts.factory.createPropertyAccessExpression(
+              ts.factory.createThis(),
+              // @ts-ignore
+              ts.factory.createIdentifier(member.name.escapedText)
+            ),
+            ts.factory.createToken(ts.SyntaxKind.EqualsToken),
+            member.initializer
+          )
+        )
+      );
+    } else {
+      // if it's not a class field that is decorated with a Stencil decorator then
+      // we just push it onto our class member list
+      updatedClassMembers.push(member);
+    }
+  }
+
+  // create or update a constructor which contains the initializing statements
+  // we created above
+  return updateConstructor(classNode, updatedClassMembers, statements);
+}
+
+/**
+ * Helper util for updating the constructor on a class declaration AST node.
+ *
+ * @param classNode the class node whose constructor we're going to be updated
```

Comment:
this reads a little funky
```suggestion
 * @param classNode the class node whose constructor that's going to be updated
```

Reply:
lol indeed it does 😅